src/app-interoperability/application-package-definition.linkml.yaml

```
@@ -0,0 +1,497 @@
+        rank: 50
+        range: integer
+      regexMatch:
+        description: If set, indicates a regular expression to use to validate the value.
+        rank: 55
+        range: string
+
+      # Boolean Validation Rules
+      # allowEmpty: #Arne: error in specification: listing of this twice
+      #   description: If true, indicates a value must be provided. Default is false if not provided.
+      #   rank: 57
+      #   range: boolean
+
+      # # Numeric Integer Validation Rules
+      # maxLength:
+      #   description: If set, indicates the minimum allowed integer value the value must have to be considered valid. 
+      #   rank: 50
+      #   range: integer
+      # regexMatch:
+      #   description: If set, indicates the maximum allowed integer value the value must have to be considered valid. 
+      #   rank: 55
+      #   range: integer
+
+      # # Numeric Double Validation Rules
+      # minValue:
+      #   description: Minimum numeric value for integer or double types.
+      #   rank: 60
+      #   range: double
+      # maxValue:
+      #   description: Maximum numeric value for integer or double types.
+      #   rank: 70
+      #   range: double
+      # minValue:
+      #   description: Minimum numeric value for integer or double types.
+      #   rank: 60
+      #   range: double
+      # maxValue:
+      #   description: Maximum numeric value for integer or double types.
+      #   rank: 70
+      #   range: double
+      
+      # # Select Validation Rules
+      # options:
+      #   description: List of acceptable values for selection inputs.
+      #   rank: 80
+      #   range: string
+      #   multivalued: true
+
+slots:
+  type:
+    description: >-
+          Indicates the components's deployment configuration. 
+          The values are `helm.v3` to indicate the component's package format is Helm version 3 
+          and `docker-compose` to indicate the component's package format is Docker Compose. 
+          When installing the application on a device supporting the Kubernetes platform all `helm.v3` components, 
+          and only `helm.v3` components, will be provided to the device in same order they are listed in the application description file. 
+          When installing the application on a device supporting docker-compose all `docker-compose` components, 
+          and only `docker-compose` components, will be provided to the device in the same order they are listed in the application description file. 
+          The device will install the components in the same order they are listed in the application description file.
+    range: string
+    required: true
+    pattern: ^(helm\.v3|docker-compose)$
+    rank: 10
+
+  components:
+    description: >-
+      Component element indicating the components to deploy when installing the application. 
+      See the [Component](#component-attributes) section below.
+    range: Component
+    multivalued: true
+    required: true
+    rank: 20
```

@arne-broering as mentioned in another comment, add `inlined: true` to specify that "objects" and not references to objects are expected. `components` is an array/list, so `inlined_as_list: true`

```suggestion
    inlined: true
    inlined_as_list: true
    rank: 20
```